pyserini/encode/_embedding_writer.py

```diff
@@ -0,0 +1,82 @@
+#
+# Pyserini: Reproducible IR research with sparse and dense representations
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import json
+import os
+
+import faiss
+import numpy as np
+
+
+class EmbeddingWriter:
```

Comment:
Not sure I like the name of this class... We don't use the term "Embedding" anywhere else. We say that encoders generate representation vectors... but it's hard to make a good name based on this.

Options I can think of: `VectorSerializer`, `VectorWriter`, `RepresentationOutput`, `RepresentationWriter`?

Do you like any of these?

Reply:
RepresentationWriter ?